torch/utils/_sympy/value_ranges.py

```diff
@@ -340,27 +361,35 @@ def convex_min_zero_map(x: Union[ExprIn, ExprVR], fn: ExprFn) -> ExprVR:
         """Fn is convex and has a minimum at 0."""
         x = ValueRanges.wrap(x)
         if 0 in x:
-            return ValueRanges(0, max(fn(x.lower), fn(x.upper)))
-        else:
-            return ValueRanges.monotone_map(x, fn)
+            upper = max(fn(x.lower), fn(x.upper))
+            if isinstance(upper, sympy.Float):
+                return ValueRanges(0.0, upper)
+            return ValueRanges(0, upper)
```

Comment:
There's something going on here, because we are apparently more sensitive to whether or not you have float zero or not in the bounds? What are you going to do if upper is sympy.oo? You're not going to correctly infer the right float or not here then...

Reply:
Add `sympy.oo` for float bounds.